server/paymentProviders/paypal/webhook.ts

```diff
@@ -114,6 +114,47 @@ async function handleSaleCompleted(req: Request): Promise<void> {
   await order.getOrCreateMembers();
 }
 
+async function handleCaptureCompleted(req: Request): Promise<void> {
+  // 1. Retrieve the order for this event
+  const capture = req.body.resource;
+  const order = await models.Order.findOne({
+    where: {
+      status: OrderStatus.NEW,
+      data: { paypalCaptureId: capture.id },
+    },
+    include: [
+      { association: 'fromCollective' },
+      { association: 'createdByUser' },
+      { association: 'collective', required: true },
+      {
+        association: 'paymentMethod',
+        required: true,
+        where: { service: 'paypal', type: 'payment' },
+      },
+    ],
+  });
+
+  if (!order) {
+    logger.debug(`No pending order found for capture ${capture.id}`);
+    return;
+  }
+
+  // 2. Validate webhook event
+  const host = await order.collective.getHostCollective();
+  const paypalAccount = await getPaypalAccount(host);
+  await validateWebhookEvent(paypalAccount, req);
+
+  // 3. Record the transaction
+  const transaction = await recordPaypalCapture(order, capture);
+  await order.update({ processedAt: new Date(), status: 'PAID' });
+
+  // 4. Send thankyou email
+  await sendThankYouEmail(order, transaction);
+
+  // 5. Register user as a member, since the transaction is not created in `processOrder`
+  await order.getOrCreateMembers();
```

Comment:
Maybe alternate the two calls, first add them as member and then send the thank you email? I normally like to send thank you email at the end since that way if the sending of email throws an exception for some reason the user will still be added as a member. 🤔

Reply:
I would say the opposite: sending the thank you email is more important than adding the members

Reply:
Great. In that case it's fine. In my mind somehow I thought adding backers is more important. 😄